content/developers/internals/data-model/usage-example.en.md

```diff
@@ -0,0 +1,275 @@
+weight: 20
+description: "OSRD data format usage example"
+---
+
+## Introduction
+
+This page gives an example of how the data formats are used to describe an infrastructure in *OSRD*.
+
+For this purpose, let's take as an example the following toy infrastructure:
+
+![Toy infrastructure schema](../svg_schemas/small_infra_schema.drawio.en.svg)
+
+{{% alert title="Tip" color="info"%}}
+In case you want to zoom in on schemas, you can click on the little editing button that appears when hovering the mouse on it.
+{{% /alert %}}
+
+In this schema we represent the overall look of the infrastructure with just the lines and the stations.
+
+This is obviously not a realistic infrastructure, but it will allow us to present the data models in a simple way.
+
+We will build the details of the infrastructure step by step, and present the objects used at each step.
+
+### The infrastructure generator
+
+In the [*OSRD* repository](https://github.com/DGEXSolutions/osrd) is a [python module](https://github.com/DGEXSolutions/osrd/tree/dev/core/examples/generated) that allows for generating infrastructures in a format understood by *OSRD*.
+
+The infrastructure we talk about in this page can be generated through the [small_infra.py](https://github.com/DGEXSolutions/osrd/blob/dev/core/examples/generated/scripts/small_infra.py) file. For more information about the generation scripts, you can check out the related [README](https://github.com/DGEXSolutions/osrd/blob/dev/core/examples/generated/README.md).
+
+## The rails
+
+### Track sections
+
+The first objects that we need to define are the `TrackSections`. Most other objects are defined in relation to the track sections we have set up.
+
+A track section is simply a portion of rail (switches not included). One can chose to divide the tracks of their infrastructure in as many track sections as they like. Here we chose to use the longest track sections possible, which means that between two switches there is always a single track section.
+
+A train runs on one track section, if you want to run trains in opposite directions at the same time and on the same line, you need to define at least two parallel track sections.
+
+For our example, we will define two tracks for the line between the West and North-East stations.
+
+Furthermore, we will add overpassing tracks at the North and Mid-West stations, to make our infrastructure a bit more realistic.
+
+We will also have three different tracks in the West station, since it's a major hub in our imaginary infrastructure.
+
+Here are the track sections that we then need:
+
+![Track sections schema](../svg_schemas/small_infra_rails.drawio.en.svg)
+
+{{% alert title="Important" color="warning"%}}
+They are represented as arrows in our schema to stress the fact that they have a **start** and an **end**. This is important because the objects related to the track section are positioned using an offset, and the offset is a distance from the **start** of the track section.
```

Comment:
```suggestion
`TrackSections` are represented as arrows in this diagram to stress the fact that they have a **start** and an **end**. This is important because the objects related to the track section are positioned using an offset, and the offset is a distance from the **start** of the track section.
```